### PR TITLE
Microfacet BRDF code refactor

### PR DESCRIPTION
**Pull request by est77**
All tests pass

### CODE REVIEW

File: src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+#include <cmath>
+
+namespace renderer
+{
+
+template <typename T>
+class MicrofacetBRDFHelper
+{
+  public:
+
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h]: Remove redundant blank line.

File: src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+// THE SOFTWARE.
+//
+
+// appleseed.renderer headers.
+#include "renderer/global/globaltypes.h"
+#include "renderer/modeling/bsdf/bsdfsample.h"
+
+// appleseed.foundation headers.
+#include "foundation/math/basis.h"
+
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h]: Add missing `foundation/math/vector.h` and possibly others.

File: src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        const MDF&  mdf,
+        const T     alpha_x,
+        const T     alpha_y,
+        const T     g_alpha_x,
+        const T     g_alpha_y,
+        FresnelFun  f,
+        BSDFSample& sample)
+    {
+        const VectorType& n = sample.get_shading_normal();
+        const T cos_on = std::min(dot(sample.get_outgoing(), n), T(1.0));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h]: Should be `foundation::dot()`.

File: src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    typedef foundation::Basis3<T>       BasisType;
+
+    template <typename MDF, typename FresnelFun>
+    static void sample(
+        const MDF&  mdf,
+        const T     alpha_x,
+        const T     alpha_y,
+        const T     g_alpha_x,
+        const T     g_alpha_y,
+        FresnelFun  f,
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h]: Rename `f` to `fresnel` maybe?

Comment (est77) [inline src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h]: I'm ok with f. It's a function object and the type makes that clear.
I did the other fixes.

Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/microfacetbrdfhelper.h]: It's just the invokation which is slightly less clear. Your call.

**Pull request merged**